### PR TITLE
Add common method for merging host_path e2e tests

### PR DESCRIPTION
**Pull request by oomichi**
**What this PR does / why we need it**:

To solve TODO on host_path.go, this adds common method for merging
2 host_path e2e tests.

**Release note**:
```NONE```

### CODE REVIEW

File: test/e2e/common/host_path.go [hunk 1]
```diff
@@ -270,3 +188,61 @@ func testPodWithHostVol(path string, source *v1.HostPathVolumeSource) *v1.Pod {
...
 	}
 }
+
+func testExistingSubPath(f *framework.Framework, fileTest bool) {
+	framework.SkipUnlessSSHKeyPresent()
+
+	var command string
+	var filePathInReader string
+
+	subPath := "sub-path"
```
Comment (oomichi) [inline test/e2e/common/host_path.go]:
```
Sep 21 01:09:31.417: INFO: Getting external IP address for ip-172-20-36-126.us-west-1.compute.internal
Sep 21 01:09:31.849: INFO: ssh admin@54.215.128.81:22: command:   echo "mount-tester new file" > /tmp/sub-path
Sep 21 01:09:31.849: INFO: ssh admin@54.215.128.81:22: stdout:    ""
Sep 21 01:09:31.849: INFO: ssh admin@54.215.128.81:22: stderr:    "bash: /tmp/sub-path: Is a directory\n"
Sep 21 01:09:31.849: INFO: ssh admin@54.215.128.81:22: exit code: 1
Sep 21 01:09:31.849: INFO: echo "mount-tester new file" > /tmp/sub-path returned non-zero
```

File: test/e2e/common/host_path.go [hunk 2]
```diff
@@ -276,3 +189,67 @@ func testPodWithHostVol(path string, source *v1.HostPathVolumeSource) *v1.Pod {
...
 }
+
+func testExistingSubPath(f *framework.Framework, fileTest bool) {
+	framework.SkipUnlessSSHKeyPresent()
+
+	var subPath string
+	var command string
+	var filePathInReader string
+
+	retryDuration := 180
```
Comment (timothysc) [inline test/e2e/common/host_path.go]: please add a consts section at the top of the file.

**Review by timothysc — APPROVED** [review]
/approve 

thanks for the cleanup. 

I'll defer to @kubernetes/sig-storage-pr-reviews for lgtm.   TBH on cursory reading inspection of these tests it's not clear to me what the difference is between the tests and why.

File: test/e2e/common/host_path.go (hunk 2 above)
Comment (oomichi) [reply test/e2e/common/host_path.go]: Thanks for yor review, @timothysc 
That is a nice point, retryDuration seems to be defined at several places in this file.
It is nice to merge them into a single const value.
OK, I will do that later with another thing.

File: test/e2e/common/host_path.go [hunk 3]
```diff
@@ -118,98 +118,11 @@ var _ = Describe("[sig-storage] HostPath", func() {
 	})
 
 	It("should support existing directory subPath", func() {
-		framework.SkipUnlessSSHKeyPresent()
```
Comment (msau42) [inline test/e2e/common/host_path.go]: I think we actually may be able to remove the subpath test cases entirely.  In test/e2e/storage/testsuites/subpath.go, we abstracted out the plugin-specific setup, which allows the subpath tests run across any plugin.

**Comment by timothysc** [comment]
@msau42 - ping.

**Comment by msau42** [comment]
I made a comment that I think we can remove this test entirely because it's already covered by test/e2e/storage/testsuites/subpath.go

**Comment by oomichi** [comment]
@msau42: Thanks for your comment.
I hope https://github.com/kubernetes/kubernetes/pull/70263 works for you instead of this.

**Comment by oomichi** [comment]
This might be unnecessary for alternative.

**Pull request closed without merging**